Framework/Kernel/src/Matrix.cpp

```diff
@@ -1613,7 +1613,7 @@ void fillFromStream(std::istream &is, Kernel::Matrix<T> &in,
 
 // Symbol definitions for common types
 template class MANTID_KERNEL_DLL Matrix<double>;
-template class MANTID_KERNEL_DLL Matrix<int>;
+template class Matrix<int>;
```

Comment:
I see. It seems that the specialization of the class template for that method has implicitly defined the symbol and inherited the visibility from the class definition. I think it might be worth putting a comment in to that effect as on first glance it looks like something is missing.